internal/activities/validate_premis_test.go

```diff
@@ -0,0 +1,154 @@
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/artefactual-sdps/temporal-activities/xmlvalidate"
+	"github.com/tonglil/buflogr"
+	"go.artefactual.dev/tools/temporal"
+	temporalsdk_activity "go.temporal.io/sdk/activity"
+	temporalsdk_interceptor "go.temporal.io/sdk/interceptor"
+	temporalsdk_testsuite "go.temporal.io/sdk/testsuite"
+	temporalsdk_worker "go.temporal.io/sdk/worker"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/fs"
+
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/activities"
+)
+
+var premisXML = `<?xml version="1.0" encoding="UTF-8"?>
+<premis:premis xmlns:premis="http://www.loc.gov/premis/v3" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.loc.gov/premis/v3 https://www.loc.gov/standards/premis/premis.xsd" version="3.0">
+  <premis:object xsi:type="premis:file">
+    <premis:objectIdentifier>
+      <premis:objectIdentifierType>uuid</premis:objectIdentifierType>
+      <premis:objectIdentifierValue>c74a85b7-919b-409e-8209-9c7ebe0e7945</premis:objectIdentifierValue>
+    </premis:objectIdentifier>
+    <premis:objectCharacteristics>
+      <premis:format>
+        <premis:formatDesignation>
+          <premis:formatName/>
+        </premis:formatDesignation>
+      </premis:format>
+    </premis:objectCharacteristics>
+    <premis:originalName>data/objects/test_transfer/content/cat.jpg</premis:originalName>
+  </premis:object>
+</premis:premis>
+`
+
+type fakeValidator struct {
+	Msg string
+	Err error
+}
+
+func (v *fakeValidator) Validate(ctx context.Context, xmlPath, xsdPath string) (string, error) {
+	return v.Msg, v.Err
+}
+
+func (v *fakeValidator) WithMsg(msg string) *fakeValidator {
+	v.Msg = msg
+	return v
+}
+
+func (v *fakeValidator) WithErr(err error) *fakeValidator {
+	v.Err = err
+	return v
+}
+
+func newFakeValidator() *fakeValidator {
+	return &fakeValidator{}
+}
+
+func TestValidatePREMIS(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name      string
+		validator xmlvalidate.XSDValidator
+		params    activities.ValidatePREMISParams
+		want      activities.ValidatePREMISResult
+		wantErr   string
+	}{
+		{
+			name:      "Validates a PREMIS file",
+			validator: xmlvalidate.NewXMLLintValidator(),
+			params: activities.ValidatePREMISParams{
+				Path: fs.NewDir(t, "enduro-test",
+					fs.WithFile("premis.xml", premisXML),
+				).Join("premis.xml"),
+			},
+		},
+		{
+			name:      "Returns a validation failure",
+			validator: newFakeValidator().WithMsg("premis.xml:12: parser error"),
+			params: activities.ValidatePREMISParams{
+				Path: fs.NewDir(t, "enduro-test",
+					fs.WithFile("premis.xml", premisXML),
+				).Join("premis.xml"),
+			},
+			want: activities.ValidatePREMISResult{
+				Failures: []string{"premis.xml does not match expected metadata requirements"},
+			},
+		},
+		{
+			name:      "Returns a file not found failure",
+			validator: newFakeValidator().WithErr(errors.New("file not found")),
```

Comment:
What are the pros and cons of having a mock object return a file not found error versus testing with a non-existent file?

Reply:
@mcantelon I think the main pro of using a mock validator is that the tests are not dependent on xmllint being installed in the test environment. Unfortunately I negated that advantage by using the real "xmllint" validator in the first test, and sure enough that test failed in CI until I added xmllint to the CI runner. :thinking: The mock validator is also faster (though I'm sure the difference isn't noticeable in these small test cases), and allows more control over the return values. 

I think the main con of using a mock validator is if there is some integration issue between the `ValidatePREMIS` activity and `xmlvalidate.XMLLintValidator` then it won't be caught by these unit tests.  Ideally integration issues would be caught by integration tests, but we don't have any integration tests for this project. :(

Reply:
Good points! Yeah, integration tests definitely seem like they'd be handy (but also potentially a lot of work to create/maintain at this stage).

Reply:
I've decided to remove the xmllint dependency in the first test here.  The deciding factor for me is that the tests should not fail if a developer runs them locally without xmllint installed.